sky/server/requests/requests.py

```diff
@@ -703,3 +712,51 @@ def set_request_cancelled(request_id: str) -> None:
     with update_request(request_id) as request_task:
         assert request_task is not None, request_id
         request_task.status = RequestStatus.CANCELLED
+
+
+@init_db
+def clean_finished_requests_with_retention(retention_seconds: int) -> None:
+    """Clean up finished requests older than the retention period.
+
+    This function removes old finished requests (SUCCEEDED, FAILED, CANCELLED)
+    from the database and cleans up their associated log files.
+
+    Args:
+        retention_seconds: Requests older than this many seconds will be
+            deleted.
+    """
+    finished_statuses = [
+        RequestStatus.SUCCEEDED, RequestStatus.FAILED, RequestStatus.CANCELLED
+    ]
+    reqs = get_request_tasks(status=finished_statuses,
+                             created_before=time.time() - retention_seconds)
+
+    for req in reqs:
+        req.log_path.unlink(missing_ok=True)
```

Comment:
wondering if we can do it in parallel / async

Reply:
Sounds good!